viewer/packages/camera-manager/src/ComboControls.ts

```diff
@@ -613,45 +643,162 @@ export default class ComboControls extends EventDispatcher {
 
     const distFromRayOrigin = -deltaDistance * ratio;
 
-    sphericalEnd.radius = radius;
+    _sphericalEnd.radius = radius;
 
-    reusableCamera.getWorldDirection(cameraDirection);
+    _reusableCamera.getWorldDirection(cameraDirection);
     cameraDirection.normalize().multiplyScalar(deltaDistance);
-    const rayDirection = raycaster.ray.direction.normalize().multiplyScalar(distFromRayOrigin);
+    const rayDirection = _raycaster.ray.direction.normalize().multiplyScalar(distFromRayOrigin);
     const targetOffset = rayDirection.add(cameraDirection);
-    targetEnd.add(targetOffset);
+
+    return targetOffset;
+  };
+
+  // Function almost equal to mapLinear except it is behaving the same as clamp outside of specifed range
+  private readonly clampedMap = (value: number, xStart: number, xEnd: number, yStart: number, yEnd: number) => {
+    if (value < xStart) value = yStart;
+    else if (value > xEnd) value = yEnd;
+    else value = MathUtils.mapLinear(value, xStart, xEnd, yStart, yEnd);
+
+    return value;
```

Comment:
```suggestion
    return MathUtils.clamp(MathUtils.mapLinear(value, xStart, xEnd, yStart, yEnd), yStart, yEnd);
```

Reply:
Seems to be wrong, because clamp function will set the `value` to `xStart`, not `yStart` when it's less than `xStart` (same for upper bound).  This was the reason why I decided to make it a separate function.